File: get_cpu_power_usage.py
#!/usr/bin/env python3
import time

ENERGY_FILE = "/sys/class/hwmon/hwmon7/energy17_input"

def get_energy():
    with open(ENERGY_FILE, "r") as f:
        return int(f.read().strip())

def main():
    initial_energy = get_energy()
    time.sleep(0.1)
    final_energy = get_energy()
    energy_diff = final_energy - initial_energy
    energy_diff_joules = energy_diff / 1_000_000
    power_usage = energy_diff_joules / 0.1

    print(f"{power_usage:.1f}")

if __name__ == "__main__":
    main()

File: get_cpu_power_usage.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
	"time"
)

const CPU_INPUT_FILE = "/sys/class/hwmon/hwmon7/energy17_input"

func getCPUEnergy() uint64 {
	data, err := ioutil.ReadFile(CPU_INPUT_FILE)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Failed to read file: %v\n", err)
		os.Exit(1)
	}
	s := strings.TrimSpace(string(data))
	energy, err := strconv.ParseUint(s, 10, 64)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Failed to parse energy value: %v\n", err)
		os.Exit(1)
	}
	return energy
}

func main() {
	initialEnergy := getCPUEnergy()
	time.Sleep(10 * time.Millisecond)
	finalEnergy := getCPUEnergy()
	energyDiff := finalEnergy - initialEnergy

	energyDiffJoules := float64(energyDiff) / 1e6
	powerUsage := energyDiffJoules / 0.01

	fmt.Printf("%.1f\n", powerUsage)
}
